src/sync-client.js

```diff
@@ -527,7 +527,7 @@ var self = {
   },
 
   generateHash: function(object) {
-    var hash = CryptoJS.SHA1(self.sortedStringify(object));
+    var hash = self.getEncryptionAdapter.sha1(self.sortedStringify(object));
```

Comment:
Also I think that we can just settle with sha1 hash method as there is no other usages of encryption library. 
So overall this may become just `self.getHashMethod(payload)`